core/src/ops/array/pad.rs

```diff
@@ -1,13 +1,27 @@
 use crate::ops::prelude::*;
 use ndarray::*;
 use num_traits::AsPrimitive;
+use std::str::FromStr;
 
 #[derive(Debug, Clone, PartialEq)]
 pub enum PadMode {
     Constant(f32),
     Reflect,
     Edge,
 }
+
+impl FromStr for PadMode {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "reflect" => Ok(PadMode::Reflect),
+            "edge" => Ok(PadMode::Edge),
+            _ => Err(())
+        }
+    }
+}
+
```

Comment:
I do not think the parser has its place in core, these strings are onnx specific.